sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ComputedProperty.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Represents a computed property definition for a Cosmos DB container.
+ */
+public final class ComputedProperty {
+    private String name;
+    private String query;
+    private JsonSerializable jsonSerializable;
+
+    /**
+     * Instantiates a new Computed properties.
+     */
+    public ComputedProperty() {
```

Comment:
Does this needs to be public? Can users create a ComputedProperty instance without name and query?